Code/GraphMol/testChirality.cpp

```diff
@@ -631,6 +632,274 @@ void testChiralityCleanup() {
+  TEST_ASSERT(mol->getBondWithIdx(2)->getBondType() == Bond::DOUBLE);
+  mol->getBondWithIdx(2)->setStereoAtoms(0, 4);
+  mol->getBondWithIdx(2)->setStereo(Bond::STEREOTRANS);
+  MolOps::setDoubleBondNeighborDirections(*mol);
+  MolOps::assignStereochemistry(*mol, true, true);
+  // std::cerr << MolToSmiles(*mol, true) << std::endl;
+  TEST_ASSERT(MolToSmiles(*mol, true) == "C/C(F)=C/[C@H](C)Cl");
+  delete mol;
+
+  // cleanIt=true, force=true should NOT remove this manual cis/trans stereo
+  // assignment
+  smi = "CC(F)=CC(Cl)C";
+  mol = SmilesToMol(smi);
+  mol->getAtomWithIdx(4)->setChiralTag(Atom::CHI_TETRAHEDRAL_CW);
+  TEST_ASSERT(mol->getBondWithIdx(2)->getBondType() == Bond::DOUBLE);
+  mol->getBondWithIdx(2)->setStereoAtoms(2, 4);
+  mol->getBondWithIdx(2)->setStereo(Bond::STEREOCIS);
+  MolOps::setDoubleBondNeighborDirections(*mol);
+  MolOps::assignStereochemistry(*mol, true, true);
+  // std::cerr << MolToSmiles(*mol, true) << std::endl;
+  TEST_ASSERT(MolToSmiles(*mol, true) == "C/C(F)=C/[C@H](C)Cl");
+  delete mol;
+
+  // cleanIt=true, force=true should clean up these manual assignments
+  smi = "FC(F)=C(Cl)Cl";
+  mol = SmilesToMol(smi);
+  TEST_ASSERT(mol->getBondWithIdx(2)->getBondType() == Bond::DOUBLE);
+  mol->getBondWithIdx(2)->setStereoAtoms(2, 4);
+  mol->getBondWithIdx(2)->setStereo(Bond::STEREOCIS);
+  MolOps::setDoubleBondNeighborDirections(*mol);
+  MolOps::assignStereochemistry(*mol, true, true);
+  TEST_ASSERT(MolToSmiles(*mol, true) == "FC(F)=C(Cl)Cl");
+  delete mol;
+
+  smi = "FC=C1CCCCC1";
+  mol = SmilesToMol(smi);
+  TEST_ASSERT(mol->getBondWithIdx(1)->getBondType() == Bond::DOUBLE);
+  mol->getBondWithIdx(1)->setStereoAtoms(0, 3);
+  mol->getBondWithIdx(1)->setStereo(Bond::STEREOTRANS);
+  MolOps::setDoubleBondNeighborDirections(*mol);
+  MolOps::assignStereochemistry(*mol, true, true);
+  TEST_ASSERT(MolToSmiles(*mol, true) == "FC=C1CCCCC1");
+  delete mol;
+
+  smi = "C1CCCCC1=CF";
+  mol = SmilesToMol(smi);
+  TEST_ASSERT(mol->getBondWithIdx(5)->getBondType() == Bond::DOUBLE);
+  mol->getBondWithIdx(5)->setStereoAtoms(4, 7);
+  mol->getBondWithIdx(5)->setStereo(Bond::STEREOCIS);
+  MolOps::setDoubleBondNeighborDirections(*mol);
+  MolOps::assignStereochemistry(*mol, true, true);
+  TEST_ASSERT(MolToSmiles(*mol, true) == "FC=C1CCCCC1");
+  delete mol;
+
+///////////////////////////
+// Pseudo-stereo test cases
+// - bond stereo dependent on other bond stereo breaking symmetry - DOESN'T
+// CURRENTLY WORK
+// - bond stereo dependent on atom stereo breaking symmetry - DOESN'T CURRENTLY
+// WORK
+// - atom stereo dependent on other atom stereo breaking symmetry - WORKS
+// - atom stereo dependent on bond stereo breaking symmetry - WORKS
+///////////////////////////
+
+// Everything ifdef'd USE_NEW_STEREOCHEMISTRY are test cases that
+// should 'hopefully' work when switching to the new_canon.h ranking
+// for stereo perception. However, making USE_NEW_STEREOCHEMISTRY
```

Comment:
I'm curious what this means. Is this saying that the morgan style canonicalization for R/S E/Z is incompatible with the current implementation?

You know my feelings about R/S E/Z identification :) I'm tempted to make a PR to have the substructure search not require it, but that's a lot of work!